install-and-configure/advanced-configuration/eks-optimized/kubecost-agentless-amp.md

```diff
@@ -0,0 +1,221 @@
+
+    ```sh
+    aws amp create-scraper --output json \
+        --alias cadvisor-scraper \
+        --source eksConfiguration="{clusterArn=$CLUSTER_ARN, securityGroupIds=[$SECURITY_GROUP_IDS],subnetIds=[$SUBNET_IDS]}" \
+        --scrape-configuration configurationBlob="$(base64 scraper-cadvisor-node-exporter.yaml|tr -d '\n')" \
+        --destination ampConfiguration="{workspaceArn=$WORKSPACE_ARN}")
+    echo $CADVSIOR_SCRAPER_OUTPUT
+    CADVSIOR_SCRAPER_ID=$(echo $CADVSIOR_SCRAPER_OUTPUT|jq -r .scraperId)
+    echo $CADVSIOR_SCRAPER_ID
+    ```
+
+8. Get the ARN of the scraper:
+
+    ```sh
+    ARN_PART=$(aws amp describe-scraper --output json --region $CLUSTER_REGION --scraper-id $CADVSIOR_SCRAPER_ID | jq -r .scraper.roleArn | cut -d'_' -f2)
+    ROLE_ARN_CADVSIOR_SCRAPER="arn:aws:iam::AWS_ACCOUNT_ID:role/AWSServiceRoleForAmazonPrometheusScraper_$ARN_PART"
+    echo $ROLE_ARN_CADVSIOR_SCRAPER
+     ```
+
+9. Add the ARN of the scraper to the kube-system/aws-auth configmap:
+
+    ```sh
+    eksctl create iamidentitymapping \
+        --cluster $CLUSTER_NAME --region $CLUSTER_REGION \
+        --arn $ROLE_ARN_CADVSIOR_SCRAPER \
+        --username aps-collector-user
+    ```
+
+10. Apply the agentless RBAC permissions:
+
+    ```sh
+    kubectl apply -f rbac.yaml
+    ```
+
+### Kubecost primary cluster installation
+
+1. Create the Kubecost namespace:
+
+    ```bash
+    kubectl create ns YOUR_NAMESPACE
+    ```
+
+1. Create the AWS IAM policy to allow Kubecost to query metrics from AMP:
+
+    ```bash
+    aws iam create-policy --policy-name kubecost-read-amp-metrics --policy-document file://iam-read-amp-metrics.json
+    ```
+
+1. (Optional) Create the AWS IAM policy to allow Kubecost to find savings in the AWS Account:
+
+    ```bash
+    aws iam create-policy --policy-name DescribeResources --policy-document file://iam-describeCloudResources.json
+    ```
+
+1. (Optional) Create the AWS IAM policy to allow Kubecost to write to find account-level tags:
+
+    ```bash
+    aws iam create-policy --policy-name OrganizationListAccountTags --policy-document file://iam-listAccounts-tags.json
+    ```
+
+1. Configure the Kubecost Service Account:
+
+    * If the following fails, be sure that IRSA is enabled on your EKS cluster. <https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html>
+
+    ```bash
+    eksctl create iamserviceaccount \
+    --name kubecost-sa \
+    --namespace YOUR_NAMESPACE \
+    --cluster $CLUSTER_NAME --region $CLUSTER_REGION \
+    --attach-policy-arn arn:aws:iam::AWS_ACCOUNT_ID:policy/kubecost-read-amp-metrics \
+    --attach-policy-arn arn:aws:iam::AWS_ACCOUNT_ID:policy/OrganizationListAccountTags \
+    --attach-policy-arn arn:aws:iam::AWS_ACCOUNT_ID:policy/DescribeResources \
+    --override-existing-serviceaccounts --approve
+    ```
+
+1. Update the place holder values such as `YOUR_CLUSTER_NAME_HERE` in [values-kubecost-primary.yaml](values-kubecost-primary.yaml)
```

Comment:
We link directly to this yaml file which doesn't exist in the repo. Is this a file they will have locally after they clone the repo? If so, there's no reason to hyperlink this